compiler/rustc_borrowck/src/diagnostics/explain_borrow.rs

```diff
@@ -162,42 +168,28 @@ impl<'tcx> BorrowExplanation<'tcx> {
 
                 match local_names[dropped_local] {
                     Some(local_name) if !local_decl.from_compiler_desugaring() => {
-                        let message = format!(
-                            "{B}borrow might be used here, when `{LOC}` is dropped \
-                             and runs the {DTOR} for {TYPE}",
-                            B = borrow_desc,
-                            LOC = local_name,
-                            TYPE = type_desc,
-                            DTOR = dtor_desc
-                        );
-                        err.span_label(body.source_info(drop_loc).span, message);
-
+                        err.subdiagnostic(UsedLaterDropped::UsedHere {
+                            borrow_desc: br_desc,
+                            local_name,
+                            type_desc: &type_desc,
+                            dtor_desc,
```

Comment:
Except english words, There are also variables passed in `dtor_desc`, which would be better:
- add a field for the diagnose struct
- make `dtor_desc` a subdiag
- some flunet magic I should go check out

Reply:
Probably a subdiagnostic.